src/hdx/scraper/utilities/reader.py

```diff
@@ -317,6 +318,16 @@ def read_hdx_metadata(
         dataset_nameinfo = datasetinfo["dataset"]
         if isinstance(dataset_nameinfo, str):
```

Comment:
sosorry, this is out of scope but I just have a question. It looks like dataset_nameinfo can be either a string or a dict - can I ask why?

Reply:
@turnerm The YAML configuration can look like what's below and this allows getting source information from different datasets depending on the HXL hashtag. It is not a use case we will need for HAPI.
 
```
    dataset:
      "#value+cbpf+funding+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gmempty+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gm0+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gm1+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gm2+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gm3+total+usd": "cbpf-allocations-and-contributions"
      "#value+cbpf+funding+gm4+total+usd": "cbpf-allocations-and-contributions"
      default_dataset: "cerf-covid-19-allocations"
```